python/tvm/script/builder/tir/for_frame.py

```diff
@@ -32,23 +32,23 @@ def __enter__(self) -> List[Var]:
         return self.vars
 
 
-def serial(start, stop, annotations) -> ForFrame:
+def serial(start, stop, annotations={}) -> ForFrame:
```

Comment:
A rule of thumb in python default arguments is: Never use anything mutable.

You may try the following example to experience how horrible python is:

```python
def func(arg={}):
    print("arg:", arg)
    arg["a"] = "b"

func()
func()
```

Reply:
```suggestion
def serial(start, stop, annotations=None) -> ForFrame:
```
So use `None` instead?

Reply:
yep